### PR TITLE
Simulations committee parent lookup issue

### PR DESCRIPTION
**Pull request by bacv**
Tree implementation returned different results when searching for child committee and then using child committee to find its parent:
```
child_comm_1, child_comm_2 = parent.children;
assert_eq(parent, child_comm_1.parent) // Failed
assert_eq(parent, child_comm_2.parent) // Succeeded
```
Added fix for this case

### CODE REVIEW

File: consensus-engine/src/overlay/branch_overlay.rs [hunk 1]
```diff
@@ -1,4 +1,8 @@
+use fraction::Fraction;
```
Comment (zeegomo) [inline consensus-engine/src/overlay/branch_overlay.rs:1]: what are the changes in this file for?

**Review by zeegomo — COMMENTED** [review]
I would suggest to add a test ensuring the layers are of the expected size (e.g. nodes at depth d are 2^d expect for the last layer where they are allowed to be < 2^d). I can add this later as well

File: consensus-engine/src/overlay/branch_overlay.rs (hunk 1 above)
Comment (bacv) [reply consensus-engine/src/overlay/branch_overlay.rs:1]: Tree overlay was updated with fractions while developing the integration tests. Branch is supposed to use same threshold logic, now both overlays use the same function for that. I can remove this for now.

Comment (zeegomo) [reply consensus-engine/src/overlay/branch_overlay.rs:1]: Yes, please do that in a separate PR

**Review by zeegomo — APPROVED** [review]

**Comment by bacv** [comment]
> I would suggest to add a test ensuring the layers are of the expected size (e.g. nodes at depth d are 2^d expect for the last layer where they are allowed to be < 2^d). I can add this later as well

Ok, will create a new PR

**Pull request merged**